src/network/network_server.cpp

```diff
@@ -429,12 +429,12 @@ NetworkRecvStatus ServerNetworkGameSocketHandler::SendCompanyInfo()
  * @param error The error to disconnect for.
  * @param reason In case of kicking a client, specifies the reason for kicking the client.
  */
-NetworkRecvStatus ServerNetworkGameSocketHandler::SendError(NetworkErrorCode error, const char *reason)
+NetworkRecvStatus ServerNetworkGameSocketHandler::SendError(NetworkErrorCode error, const std::string &reason)
 {
 	Packet *p = new Packet(PACKET_SERVER_ERROR);
 
 	p->Send_uint8(error);
-	if (reason != nullptr) p->Send_string(reason);
+	if (!reason.empty()) p->Send_string(reason);
```

Comment:
We do what now? Oof .. nothing to fix in this PR, but sending fields like this optional is .. euh .. weird.

Reply:
Agreed, it is weird. But the other side in network_client.cpp is even weirder. And... it is part of the happy 8 (previously 6) packets that must remain in order and mostly retain their format. So this is mostly to not break that.